Mathlib/Combinatorics/SimpleGraph/ConcreteColorings.lean

```diff
@@ -48,4 +48,66 @@ theorem chromaticNumber_pathGraph (n : ℕ) (h : 2 ≤ n) :
   · simpa only [pathGraph_two_eq_top, chromaticNumber_top] using
       hc.chromaticNumber_mono_of_embedding (pathGraph_two_embedding n h)
 
+/-- In a bicolored graph colors alternate on every path -/
+theorem pathGraph_Hom_coloring {α} (G : SimpleGraph α) (c : G.Coloring Prop) {n : ℕ} (hn : 1 ≤ n)
+    (hom : pathGraph n →g G) (hc0 : c (hom ⟨0, hn⟩)) (u : Fin n) :
```

Comment:
Does it work in your application to do
```suggestion
theorem pathGraph_Hom_coloring {α} (G : SimpleGraph α) (c : G.Coloring Prop) {n : ℕ}
    (hom : pathGraph (n + 1) →g G) (hc0 : c (hom ⊥)) (u : Fin (n + 1)) :
```
It would be more idiomatic. If you can't, note that you can at least do
```suggestion
theorem pathGraph_Hom_coloring {α} (G : SimpleGraph α) (c : G.Coloring Prop) {n : ℕ}
    {u : Fin n} (hom : pathGraph n →g G) (hc0 : c (hom ⟨0, bot_le.trans_lt u.2⟩)) :
```